In [36]:
import pandas as pd
df = pd.read_csv('data/facebook_ad_sample.csv')

In [37]:
df.head()

ad_id  xyz_campaign_id  fb_campaign_id    age gender    interest  \
0  708746              916          103916  30-34      M  intrest.15   
1  708749              916          103917  30-34      M  intrest.16   
2  708771              916          103920  30-34      M  intrest.20   
3  708815              916          103928  30-34      M  intrest.28   
4  708818              916          103928  30-34      M  intrest.28   

   Impressions  Clicks  Spent  Total_Conversion  Approved_Conversion     CTR  \
0         7350       1   1.43                 2                    1  0.0136   
1        17861       2   1.82                 2                    0  0.0112   
2          693       0   0.00                 1                    0  0.0000   
3         4259       1   1.25                 1                    0  0.0235   
4         4133       1   1.29                 1                    1  0.0242   

    CPC  
0  1.43  
1  0.91  
2   NaN  
3  1.25  
4  1.29

In [38]:
df['xyz_campaign_id'].unique()

array([ 916,  936, 1178])

#분석

### 칼럼별 데이터 개요
	1.	ad_id: 각 광고에 할당된 고유 번호. 분석 시 유용하지 않은 데이터.
	2.	xyz_campaign_id: XYZ 회사에서 특정 캠페인을 식별하기 위한 번호. 916, 936, 1178 세가지 번호가 있는 것으로 보아 3개의 캠페인이 있다고 할 수 있음. 광고의 시안이라고 볼 수 있다.
	3.	fb_campaign_id: 페이스북 플랫폼에서 해당 캠페인을 식별하기 위한 번호.
	4.	age: 광고가 타겟팅한 나이 범위.
	5.	gender: 광고가 타겟팅한 성별 (M: 남성, F: 여성).
	6.	interest: 광고가 타겟팅한 관심사 범주.
	7.	Impressions: 광고가 화면에 노출된 총 횟수.
	8.	Clicks: 광고가 클릭된 총 횟수.
	9.	Spent: 광고비로 사용된 총 금액.
	10.	Total_Conversion: 광고 클릭 후 사용자가 수행한 전환 횟수.
	11.	Approved_Conversion: 승인된 전환 횟수, 유효 전환 횟수.
	12.	CTR (Click Through Rate): 클릭률.
	13.	CPC (Cost Per Click): 클릭당 비용.

### 기본 데이터 분석
	- 어떤 광고 시안을 사용할지 결정해야 한다.
	- 소비자 타겟을 결정하기 위해 어떤 연령, 어떤 성별, 어떤 관심사의 소비자를 타겟할지 결정해야한다.

	- 광고 시안별 클릭률 / 유효 전환률 분석
	- 소비자의 관심사 타겟별 클릭률 / 유효 전환률 분석
	- 소비자의 연령별 클릭률 / 유효 전환률 분석
	- 소비자의 성별별 클릭률 / 유효 전환률 분석

In [39]:
#광고 시안별 클릭률 분석

df.groupby(['xyz_campaign_id'])[['CTR']].mean()

#실행 결과 광고 시안별 클릭률은 차이가 없음을 알 수 있음

CTR
xyz_campaign_id          
916              0.018011
936              0.016502
1178             0.016220

In [40]:
# 광고 시안별 유효 전환률 분석

#df.head()
df.groupby(['xyz_campaign_id'])[['Approved_Conversion']].mean()

# 실행결과 1178번 시안의 전환률이 높은 것을 알 수 있음


Approved_Conversion
xyz_campaign_id                     
916                         0.444444
936                         0.394397
1178                        1.395200

In [41]:
# 광고 시안별 비용 대비 전환률 분석

conv_spent = df.groupby(['xyz_campaign_id'])[['Approved_Conversion', 'Spent']].sum()
conv_spent

Approved_Conversion         Spent
xyz_campaign_id                                   
916                               24    149.710001
936                              183   2893.369999
1178                             872  55662.149959

In [42]:
conv_spent['conv_spent'] = conv_spent['Approved_Conversion'] / conv_spent['Spent']
conv_spent

# 분석결과 지출 비용 대비 전환률은 1178에서 가장 낮은 것을 알 수 있음. 

Approved_Conversion         Spent  conv_spent
xyz_campaign_id                                               
916                               24    149.710001    0.160310
936                              183   2893.369999    0.063248
1178                             872  55662.149959    0.015666

In [34]:
# 소비자의 관심사 타겟별 클릭률 / 유효 전환률 분석.
df.head()
df.groupby(['interest'])[['CTR', 'CPC']].mean()
# 유의한 차이가 없음

CTR       CPC
interest                       
intrest.10   0.017455  1.487391
intrest.100  0.014333  1.456667
intrest.101  0.014286  1.465714
intrest.102  0.011514  1.412857
intrest.103  0.014580  1.460000
intrest.104  0.016040  1.472000
intrest.105  0.014729  1.448571
intrest.106  0.019160  1.452000
intrest.107  0.012650  1.436250
intrest.108  0.012414  1.467143
intrest.109  0.017883  1.443333
intrest.110  0.013713  1.426250
intrest.111  0.015383  1.430000
intrest.112  0.010729  1.477143
intrest.113  0.009783  1.481667
intrest.114  0.014480  1.502000
intrest.15   0.013135  1.497179
intrest.16   0.014344  1.455431
intrest.18   0.014835  1.510294
intrest.19   0.017906  1.575000
intrest.2    0.017236  1.491579
intrest.20   0.020059  1.501282
intrest.21   0.014208  1.616400
intrest.22   0.020482  1.418065
intrest.23   0.015183  1.567778
intrest.24   0.024700  1.538182
intrest.25   0.021481  1.421250
intrest.26   0.017441  1.479000
intrest.27   0.020562  1.589231
intrest.28   0.019602  1.464419
intrest.29   0.015234  1.450328
intrest.30   0.017108  1.532632
intrest.31   0.013932  1.532941
intrest.32   0.013652  1.649583
intrest.36   0.012481  1.584000
intrest.63   0.017048  1.516053
intrest.64   0.015529  1.524722
intrest.65   0.018263  1.452000
intrest.66   0.014155  1.620000
intrest.7    0.013675  1.479412

In [47]:
# 소비자의 연령 타겟별 클릭률 / 유효 전환률 분석.
df.head()
df.groupby(['age'])[['CTR', 'CPC', 'Approved_Conversion']].mean()

CTR       CPC  Approved_Conversion
age                                           
30-34  0.011623  1.506059             1.159624
35-39  0.016256  1.518104             0.834677
40-44  0.020046  1.475775             0.809524
45-49  0.021523  1.492597             0.803089

In [48]:
conv_spent = df.groupby(['age'])[['Approved_Conversion', 'Spent']].sum()
conv_spent['conv_spent'] = conv_spent['Approved_Conversion'] / conv_spent['Spent']
conv_spent
#클릭률은 40대에서 높지만, 전환률 및 비용대비 전환률은 30-34에서 가장 높음

Approved_Conversion         Spent  conv_spent
age                                                 
30-34                  494  15252.399986    0.032388
35-39                  207  11112.429994    0.018628
40-44                  170  11589.729981    0.014668
45-49                  208  20750.669997    0.010024

In [51]:
# 소비자의 성별 타겟별 클릭률 / 유효 전환률 분석.
df.head()
df.groupby(['gender'])[['CTR', 'CPC', 'Approved_Conversion']].mean()

CTR       CPC  Approved_Conversion
gender                                         
F       0.019389  1.428877             0.898367
M       0.013655  1.568436             0.986486

In [54]:
conv_spent = df.groupby(['gender'])[['Approved_Conversion', 'Spent']].sum()
conv_spent['conv_spent'] = conv_spent['Approved_Conversion'] / conv_spent['Spent']
conv_spent
#비용대비 전환률은 여성이 우세함.

Approved_Conversion         Spent  conv_spent
gender                                               
F                       495  34502.619963    0.014347
M                       584  24202.609995    0.024130

In [55]:
# 소비자의 성별연령 타겟별 클릭률 / 유효 전환률 분석.
df.head()
df.groupby(['gender', 'age'])[['CTR', 'CPC', 'Approved_Conversion']].mean()

CTR       CPC  Approved_Conversion
gender age                                           
F      30-34  0.013546  1.439128             0.989848
       35-39  0.019274  1.418172             0.871560
       40-44  0.022106  1.408947             0.869159
       45-49  0.025717  1.439683             0.811594
M      30-34  0.009970  1.569177             1.305677
       35-39  0.013888  1.596864             0.805755
       40-44  0.017906  1.544783             0.747573
       45-49  0.016740  1.556095             0.793388

In [58]:
conv_spent = df.groupby(['gender', 'age'])[['Approved_Conversion', 'Spent']].sum()
conv_spent['conv_spent'] = conv_spent['Approved_Conversion'] / conv_spent['Spent']
conv_spent.sort_values(by='conv_spent', ascending=False)
#비용대비 전환률은 30-34 남성, 30-34 여성, 35-39 남성에서 가장 높음

Approved_Conversion         Spent  conv_spent
gender age                                                 
M      30-34                  299   7640.919991    0.039131
F      30-34                  195   7611.479995    0.025619
M      35-39                  112   5051.080003    0.022173
       40-44                   77   4193.149997    0.018363
F      35-39                   95   6061.349992    0.015673
M      45-49                   96   7317.460004    0.013119
F      40-44                   93   7396.579984    0.012573
       45-49                  112  13433.209993    0.008338

## 분석결과

- 분석결과 1번 시안을 사용해서 30-34 남성, 30-34 여성, 35-39 남성을 타겟하는 것이 가장 비용 대비 전환 효율이 가장 높음을 알 수 있음.